pyteal/ast/frame_test.py

```diff
@@ -105,3 +113,180 @@ def test_dupn_invalid():
 
     with pytest.raises(pt.TealInputError):
         DupN(pt.Int(1), 1).__teal__(avm7Options)
+
+
+def test_local_type_segment_invalid():
```

Comment:
would be nice to assert a bit about the expected error. For example, the last case could be rewritten as:

```python
    with pytest.raises(pt.TealInternalError) as tie:
        LocalTypeSegment(pt.TealType.none, 2)
    assert "frame memory layout must be typed" in str(tie)
```

Reply:
good point, just add all `tie` error message check